presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PrestoUpgrader.java

```diff
@@ -13,57 +13,56 @@
  */
 package com.teradata.prestomanager.agent;
 
-import com.teradata.prestomanager.agent.api.PackageAPI.PackageType;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 import java.io.File;
 import java.io.IOException;
 import java.net.URL;
+import java.nio.file.Paths;
 
+import static com.google.common.io.MoreFiles.deleteDirectoryContents;
 import static com.teradata.prestomanager.agent.CommandExecutor.executeCommand;
+import static com.teradata.prestomanager.agent.PackageApiUtils.checkRpmPackage;
+import static com.teradata.prestomanager.agent.PackageApiUtils.fetchFileFromUrl;
 import static java.io.File.createTempFile;
 import static java.lang.String.format;
 import static java.util.Objects.requireNonNull;
-import static org.apache.commons.io.FileUtils.copyURLToFile;
 
 public class PrestoUpgrader
         implements PrestoCommand
 {
     private static final Logger LOGGER = LogManager.getLogger(PrestoUpgrader.class);
-    private static final int CONNECTION_TIMEOUT = 1000000;
-    private static final int READ_TIMEOUT = 10000;
+    private static final int SUBPROCESS_TIMEOUT = 150;
+    private static final String CONFIG_DIR = "/etc/presto";
 
     private final PackageType packageType;
     private final URL urlToFetchPackage;
     private final boolean disableDependencyChecking;
+    private final boolean retainConfigs;
 
-    public PrestoUpgrader(PackageType packageType, URL urlToFetchPackage, boolean disableDependencyChecking)
+    public PrestoUpgrader(PackageType packageType, URL urlToFetchPackage, boolean disableDependencyChecking, boolean retainConfigs)
```

Comment:
I suspect that it would be a better design decision to not pass the package type here, and instead use a different class to manage tarball installations.

Also, `urlToFetchPackage` could be shortened to `packageUrl`. (And I would reverse `disableDependencyChecking` to `checkDependencies` for both brevity and to avoid the double negative.)

```java
/*
 * @param packageType The type of package to install.
 * @param packageUrl The URL from which to retrieve Presto.
 * @param checkDependencies Whether to check RPM dependencies before an RPM upgrade.
 * @param retainConfig Whether to preserve the existing configuration files.
 */
```